arangod/Metrics/ClusterMetricsFeature.h

```diff
@@ -0,0 +1,99 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2022 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Valery Mironov
+////////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+
+#include "ApplicationFeatures/ApplicationFeature.h"
+#include "ProgramOptions/ProgramOptions.h"
+#include "Metrics/Parse.h"
+#include "Metrics/MetricKey.h"
+#include "Scheduler/SchedulerFeature.h"
+#include "Containers/FlatHashMap.h"
+
+#include <chrono>
+#include <memory>
+#include <string>
+#include <vector>
+#include <shared_mutex>
+#include <variant>
+
+namespace arangodb::metrics {
+
+class ClusterMetricsFeature final
+    : public application_features::ApplicationFeature {
+ public:
+  using MetricValue = std::variant<uint64_t>;
+  using MetricKey = std::pair<std::string, std::string>;
+  using CoordinatorMetrics = std::map<MetricKey, MetricValue>;
```

Comment:
Maybe replace to `FlatHashMap<std::string, FlatHashMap<std::string, MetricValue>>`?

Reply:
Maybe later